snyk/client.py

```diff
@@ -108,6 +175,47 @@ def delete(self, path: str) -> requests.Response:
             raise SnykHTTPError(resp)
         return resp
 
+    def get_v3_pages(self, path: str, params: dict = {}) -> List:
+        """
+        Helper function to collect paginated responses from the V3 API into a single
+        list.
+
+        This collects the "data" list from the first reponse and then appends the
+        any further "data" lists if a next link is found in the links field.
+        """
+
+        # this is a raw primative but a higher level module might want something that does an
+        # arbitrary path + origin=foo + limit=100 url construction instead before being sent here
+
+        limit = params["limit"]
+
+        data = list()
+
+        page = self.get(path, params).json()
+
+        data.extend(page["data"])
+
+        while "next" in page["links"].keys():
+            logger.debug(f"GET_V3_PAGES: Another link exists: {page['links']['next']}")
+
+            next_url = urllib.parse.urlsplit(page["links"]["next"])
+            query = urllib.parse.parse_qs(next_url.query)
+
+            for k, v in query.items():
+                params[k] = v
+
+            params["limit"] = limit
+
+            page = self.get(next_url.path, params).json()
```

Comment:
Why have this return a list of JSON? That feels different to the rest of the API, and limits the utility of this method?